### PR TITLE
[LIB-783] - Correct endpoints check methods

### PR DESCRIPTION
**Pull request by ilu2112**
@opalczynski Here we go with PR. Please check if it's fine. If yes then I will adapt other models using the same approach.

### CODE REVIEW

File: syncano/models/incentives.py [hunk 1]
```diff
@@ -267,8 +267,7 @@ def run(self, cache_key=None, **payload):
         if self.is_new():
             raise SyncanoValidationError('Method allowed only on existing model.')
 
-        properties = self.get_endpoint_data()
-        endpoint = self._meta.resolve_endpoint('run', properties)
```
Comment (opalczynski) [inline syncano/models/incentives.py]: In:
```
    def resolve_endpoint(self, name, properties):
        endpoint = self.get_endpoint(name)

        for name in endpoint['properties']:
            if name not in properties:
                raise SyncanoValueError('Request property "{0}" is required.'.format(name))

        return endpoint['path'].format(**properties)
```

is `get_endpoint` which is:
```
    def get_endpoint(self, name):
        if name not in self.endpoints:
            raise SyncanoValueError('Invalid path name: "{0}".'.format(name))
        return self.endpoints[name]
```
Maybe we should add check in `resolve_endpoint` for that? Would be much cleaner. What do you think?

Comment (ilu2112) [inline syncano/models/incentives.py]: @opalczynski Fine :)

**Comment by ilu2112** [comment]
@opalczynski Please check this now :)

File: syncano/models/options.py [hunk 2]
```diff
@@ -133,15 +133,23 @@ def get_endpoint_methods(self, name):
...
-        for name in endpoint['properties']:
-            if name not in properties:
-                raise SyncanoValueError('Request property "{0}" is required.'.format(name))
+        for endpoint_name in endpoint['properties']:
+            if endpoint_name not in properties:
+                raise SyncanoValueError('Request property "{0}" is required.'.format(endpoint_name))
 
         return endpoint['path'].format(**properties)
 
+    def is_http_method_available_for_endpoint(self, http_method_name, endpoint_name):
```
Comment (opalczynski) [inline syncano/models/options.py]: `is_http_method_available_for_endpoint` -> `is_http_method_available` is sufficient

File: syncano/models/incentives.py [hunk 3]
```diff
@@ -279,7 +280,7 @@ def run(self, cache_key=None, **payload):
         if params:
             kwargs.update({'params': params})
 
-        response = connection.request('POST', endpoint, **kwargs)
+        response = connection.request(endpoint_http_method, endpoint, **kwargs)
```
Comment (opalczynski) [inline syncano/models/incentives.py]: `http_method` will be enough

Comment (opalczynski) [inline syncano/models/incentives.py]: Above also.

File: syncano/models/options.py [hunk 4]
```diff
@@ -133,15 +133,23 @@ def get_endpoint_methods(self, name):
         endpoint = self.get_endpoint(name)
         return endpoint['methods']
 
-    def resolve_endpoint(self, name, properties):
-        endpoint = self.get_endpoint(name)
+    def resolve_endpoint(self, endpoint_name, properties, http_method=None):
+        if http_method and not self.is_http_method_available_for_endpoint(http_method, endpoint_name):
+            raise SyncanoValidationError(
+                'HTTP method {0} not allowed for endpoint name "{1}".'.format(http_method, endpoint_name)
```
Comment (opalczynski) [inline syncano/models/options.py]: `HTTP method {0} not allowed for endpoint name` -> `HTTP method {0} not allowed for endpoint `

**Comment by opalczynski** [comment]
Please add some tests;

**Comment by ilu2112** [comment]
Yes, tests will be there. I wanted to know if the approach is correct.

**Comment by ilu2112** [comment]
@opalczynski it's ready :)

File: syncano/models/channels.py [hunk 5]
```diff
@@ -138,7 +138,7 @@ class Meta:
 
     def poll(self, room=None, last_id=None, callback=None, error=None, timeout=None):
         properties = self.get_endpoint_data()
-        endpoint = self._meta.resolve_endpoint('poll', properties)
+        endpoint = self._meta.resolve_endpoint('poll', properties, 'GET')
```
Comment (opalczynski) [inline syncano/models/channels.py]: :P you're using in all previous one the pattern:
http_method = 'GET'

self._meta.resolve(..., http_method)

maybe just add:
`endpoint = self._meta.resolve_endpoint('poll', properties, http_method='GET')`

or make http_method='GET' as default

**Comment by opalczynski** [comment]
This is nice. One small comment. LGTM.

**Comment by opalczynski** [comment]
++ 
just wait for the green CI :)

**Pull request merged**